### DISCUSSION TITLE
How to compute the delay between 2 columns containing dates

### DISCUSSION THREAD

**Discussion by leprince-martine** [fbaligand/kibana-enhanced-table#166] [Q&A] [answered]
Hello

We have just installed "Enhanced Table" and "Document Table" compatible version Kibana 7.10. 
I'm trying to use "Document Table", which seems to me to be a very interesting feature.

I want to calculate the delay between 2 'date type' columns, which are in columns 2 ('SBB Date') and 3 ('PDP Date'). 
I simply try col2-col3, the result column remains empty. I also try col['SBB Date'] - col['PDP Date'], with no success.

I also try to use parseDate(col2) - parseDate(col3), I have the error :
"Document Table Error
undefined variable: parseDate, invalid expression in computed column: parseDate(col2)"

Do you have an idea of the problem?

Best regards.
Martine Leprince, from Thales Alenia Space, Cannes

**Comment by fbaligand** [comment, marked as answer]
Hi,

First, to compute the delay between 2 dates, you do the right formula: `parseDate(col2) - parseDate(col3)`
Then I reproduced your issue, when upgrading enhanced-table v1.10.1 to v1.11.0.
And I fixed the issue, by totally removing browser cache.
I don't mean "Shift+F5" or "Ctrl+F5", but total cache removal in browser.

Hope it will fix your issue,
Et bonne nuit :)

**Comment by leprince-martine** [comment]
Hi Fabien,

In fact, we have installed the v1.10.1 version.
I understand from the info on the Release page that the parseDate function and the timeRange variable only work from version v1.11.0 onwards, so I ask our installer to upgrade your plug-in to v1.11.0.

et merci pour votre réactivité ☺

Cordialement
Martine Leprince.
De : Fabien Baligand [mailto:notifications@github.com]
Envoyé : mardi 5 janvier 2021 01:29
À : fbaligand/kibana-enhanced-table <kibana-enhanced-table@noreply.github.com>
Cc : LEPRINCE Martine <martine.leprince@thalesaleniaspace.com>; Author <author@noreply.github.com>
Objet : Re: [fbaligand/kibana-enhanced-table] How to compute the delay between 2 columns containing dates (#166)


Hi,

First, to compute the delay between 2 dates, you do the right formula: parseDate(col2) - parseDate(col3)
Then I reproduced your issue, when upgrading enhanced-table v1.10.1 to v1.11.0.
And I fixed the issue, by totally removing browser cache.
I don't mean "Shift+F5" or "Ctrl+F5", but total cache removal in browser.

Hope it will fix your issue,
Et bonne nuit :)

**Comment by fbaligand** [comment]
Nice 👍

**Comment by leprince-martine** [comment]
Hi Fabien

We have installed to new version v1.11.0.
Firsts tests are OK.

Thank you very very much ☺.


De : LEPRINCE Martine
Envoyé : mardi 5 janvier 2021 09:26
À : 'fbaligand/kibana-enhanced-table' <reply@reply.github.com>
Objet : RE: [fbaligand/kibana-enhanced-table] How to compute the delay between 2 columns containing dates (#166)

Hi Fabien,

In fact, we have installed the v1.10.1 version.
I understand from the info on the Release page that the parseDate function and the timeRange variable only work from version v1.11.0 onwards, so I ask our installer to upgrade your plug-in to v1.11.0.

et merci pour votre réactivité ☺

Cordialement
Martine Leprince.
De : Fabien Baligand [mailto:notifications@github.com]
Envoyé : mardi 5 janvier 2021 01:29
À : fbaligand/kibana-enhanced-table <kibana-enhanced-table@noreply.github.com<mailto:kibana-enhanced-table@noreply.github.com>>
Cc : LEPRINCE Martine <martine.leprince@thalesaleniaspace.com<mailto:martine.leprince@thalesaleniaspace.com>>; Author <author@noreply.github.com<mailto:author@noreply.github.com>>
Objet : Re: [fbaligand/kibana-enhanced-table] How to compute the delay between 2 columns containing dates (#166)


Hi,

First, to compute the delay between 2 dates, you do the right formula: parseDate(col2) - parseDate(col3)
Then I reproduced your issue, when upgrading enhanced-table v1.10.1 to v1.11.0.
And I fixed the issue, by totally removing browser cache.
I don't mean "Shift+F5" or "Ctrl+F5", but total cache removal in browser.

Hope it will fix your issue,
Et bonne nuit :)

**Comment by fbaligand** [comment]
Great!
